src/iris/api.py

```diff
@@ -1092,6 +1098,21 @@ def on_get(self, req, resp):
 
         query = plan_query % ', '.join(plan_columns[f] for f in fields)
 
+        connection = db.engine.raw_connection()
+        cursor = connection.cursor(db.ss_dict_cursor)
+
+        plan_target = req.params.pop('target', None)
+        # search for plans which have steps that target a specific user
+        if plan_target is not None:
+            try:
+                cursor.execute('SELECT `id` FROM `target` WHERE `name` = %s', plan_target)
+                plan_target_id = cursor.fetchone()['id']
+            except Exception:
+                raise HTTPBadRequest('Invalid provided target.')
```

Comment:
Instead of this, raise `HTTPBadRequest` if `cursor.fetchone()` returns `None`. We shouldn't hide possible DB errors with a 400. Although this may go away if you use `gen_where_filter_clause`

Reply:
changed to use same dbquery as all the search calls